extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesPeonClient.java

```diff
@@ -42,12 +48,22 @@ public class KubernetesPeonClient
   private final KubernetesClientApi clientApi;
   private final String namespace;
   private final boolean debugJobs;
+  private final TaskAdapter adapter;
+  private final ServiceEmitter emitter;
 
-  public KubernetesPeonClient(KubernetesClientApi clientApi, String namespace, boolean debugJobs)
+  public KubernetesPeonClient(
+      KubernetesClientApi clientApi,
+      String namespace,
+      boolean debugJobs,
+      TaskAdapter adapter,
+      ServiceEmitter emitter
+  )
   {
     this.clientApi = clientApi;
     this.namespace = namespace;
     this.debugJobs = debugJobs;
+    this.adapter = adapter;
+    this.emitter = emitter;
   }
 
   public Pod launchPeonJobAndWaitForStart(Job job, long howLong, TimeUnit timeUnit)
```

Comment:
The job object that is being passed in here is calculated by using the adapter to convert a task to the Job in `KubernetesTaskRunner#doTask`. And then later we are using the adapter to convert it back to a task.

Would it be cleaner to just pass in the Task object here instead?

Reply:
i am also hoping to deprecate toTask(Job) in this pr: https://github.com/apache/druid/pull/14802/files to allow us to pull tasks instead of storing them as a env variable